src/software/pipeline/main_incrementalSfM.cpp

```diff
@@ -282,25 +282,32 @@ int main(int argc, char **argv)
 
   if(minNbObservationsForTriangulation < 2)
   {
-      // allows to use to the old triangulatation algorithm (using 2 views only) during resection.
-      minNbObservationsForTriangulation = 0;
-      // ALICEVISION_LOG_ERROR("The value associated to the argument '--minNbObservationsForTriangulation' must be >= 2 ");
-      // return EXIT_FAILURE;
+    // allows to use to the old triangulatation algorithm (using 2 views only) during resection.
+    minNbObservationsForTriangulation = 0;
+    // ALICEVISION_LOG_ERROR("The value associated to the argument '--minNbObservationsForTriangulation' must be >= 2 ");
+    // return EXIT_FAILURE;
   }
   sfmEngine.setNbOfObservationsForTriangulation(minNbObservationsForTriangulation);
 
-  // handle Initial pair parameter
-  if(!initialPairString.first.empty() && !initialPairString.second.empty())
+  // handle initial pair parameter
+  if(!initialPairString.first.empty() || !initialPairString.second.empty())
   {
+    bool isInitialPairCorrect = true;
+    Pair initialPairIndex;
+
     if(initialPairString.first == initialPairString.second)
     {
       ALICEVISION_LOG_ERROR("Invalid image names. You cannot use the same image to initialize a pair.");
       return EXIT_FAILURE;
     }
 
-    Pair initialPairIndex;
-    if(!retrieveViewIdFromImageName(sfmData, initialPairString.first, initialPairIndex.first) ||
-       !retrieveViewIdFromImageName(sfmData, initialPairString.second, initialPairIndex.second))
+    if(!initialPairString.first.empty() && !retrieveViewIdFromImageName(sfmData, initialPairString.first, initialPairIndex.first))
```

Comment:
You could log inside the "if", so if only of the 2 inputs is invalid, we directly get the information.
Same in the 2nd "if".